include/GafferImage/TypeIds.h

```diff
@@ -113,6 +113,9 @@ enum TypeId
 	MixTypeId = 110817,
 	CollectImagesTypeId = 110818,
 	DeleteImageContextVariablesTypeId = 110819,
+	RankFilterTypeId = 110820,
```

Comment:
One small comment - it looks like none of the preceding commits in this branch would compile without this later commit. Ideally all given commits would compile (and the tests would pass), so that we can use `git bisect` effectively when hunting down bugs. No need to change anything now, just something that would make life a little bit easier in future...